PHPCompatibility/Tests/Syntax/NewDynamicClassConstantFetchUnitTest.php

```diff
@@ -0,0 +1,130 @@
+<?php
+/**
+ * PHPCompatibility, an external standard for PHP_CodeSniffer.
+ *
+ * @package   PHPCompatibility
+ * @copyright 2012-2020 PHPCompatibility Contributors
+ * @license   https://opensource.org/licenses/LGPL-3.0 LGPL3
+ * @link      https://github.com/PHPCompatibility/PHPCompatibility
+ */
+
+namespace PHPCompatibility\Tests\Syntax;
+
+use PHPCompatibility\Tests\BaseSniffTestCase;
+
+/**
+ * Test the NewDynamicClassConstantFetch sniff.
+ *
+ * @group newDynamicClassConstantFetch
+ * @group syntax
+ *
+ * @covers \PHPCompatibility\Sniffs\Syntax\NewDynamicClassConstantFetchSniff
+ *
+ * @since 10.0.0
+ */
+final class NewDynamicClassConstantFetchUnitTest extends BaseSniffTestCase
+{
+
+    /**
+     * Ensure a warning message when found syntax on earlier versions.
+     *
+     * @dataProvider dataDynamicClassConstantFetch
+     *
+     * @param int $line The line number.
+     *
+     * @return void
+     */
+    public function testDynamicClassConstantFetch($line)
+    {
+        $file = $this->sniffFile(__FILE__, '8.2');
+        $this->assertError($file, $line, 'Dynamic class constant fetch is not available in PHP 8.2 or earlier.');
+    }
+
+    /**
+     * Data provider.
+     *
+     * @return array
```

Comment:
Nitpick
```suggestion
     * @return array<array<int>>
```

Open question: why is the `@see` tag below the `@return` tag ? Force of habit ?

In this codebase, it will normally be above the `@param`/`@return` tags, with the reason being that the `@see` is part of the _long description_ of the function.

Reply:
There was no intention behind it. Not sure if I just saw it elsewhere or moved things around until this was the end result. I don't have a preference.